### PR TITLE
Updates to pkgconfig files

### PR DESCRIPTION
**Pull request by pshriwise**
This PR addresses #2679. 

With these changes Pkgconfig files are only installed for build methods which are enabled. 

Additionally, the input file used to configure the `libmesh.pc` file is chosen based on the enabled build methods as well with a preference order of `opt`, `dbg`, `devel`, `prof`, `oprof`. This ensures that the `libmesh.pc` is always valid and can be used by external applications consistently.

### CODE REVIEW

**Comment by pshriwise** [comment]
Wondering if `devel` should come before `dbg` in order of preference for the `libmesh.pc` file. Any thoughts @jwpeterson?

**Comment by jwpeterson** [comment]
You'll have to run `./bootstrap` in the top level of the repo and commit the results in order for your changes to make their way into the "generated" autoconf files. We do this so that regular users just have to run configure/make/make install and don't have to worry about bootstrapping. And you'll want to be sure and use the version of the autotools that comes with libmesh for this... there are some instructions [here](https://github.com/libMesh/libmesh/wiki/Updating-configure-tests%2C-Makefiles%2C-etc.)

> Wondering if devel should come before dbg in order of preference

Are you asking for the case when the user does not build opt but does build both debug and devel methods? I'd say dbg, but I don't feel too strongly about it either way.

**Comment by pshriwise** [comment]
Took care of the bootstrapping I think. Let me know if something went wrong though!

>Are you asking for the case when the user does not build opt but does build both debug and devel methods? I'd say dbg, but I don't feel too strongly about it either way.

Yeah that's what I had in mind. I was wondering if performance might take precedence there, but I think you're right.

**Comment by jwpeterson** [comment]
Thanks, since our main CI is down at the moment, it might be a little while before this gets merged. 

By the way, if I wanted to test the pkg-config files on my system, what exactly would I run? I guess something like 
```
pkg-config --cflags --libs $LIBMESH_DIR/etc/libmesh/libmesh.pc
```
is sufficient? I gather that by placing that directory in `$PKG_CONFIG_PATH` I could shorten this to:
```
pkg-config --cflags --libs libmesh
```
Do you use this in your project's Makefiles or something?

**Comment by pshriwise** [comment]
Sounds good. No big rush. It'd be great to have this in before the next release though. :crossed_fingers: 

That's what I've been using to test it, yeah. I've been using CMake's `pkg_check_modules` primarily, but I have other test projects using Makefile where it will be nice to be able to always rely on the `libmesh.pc` file too.

**Comment by jwpeterson** [comment]
I had a chance to test this in a build directory where only dbg mode is built and it seems to work fine. I also verified that re-running ./bootstrap produces no changes. So I think it's probably safe to merge even though CI hasn't had a chance to run yet... Thanks @pshriwise!
